packages/core/src/lib/sign-in-experience.ts

```diff
@@ -21,12 +23,44 @@ export const validateTermsOfUse = (termsOfUse: TermsOfUse) => {
   );
 };
 
-export const validateSignInMethods = (signInMethods: SignInMethods) => {
+const isEnabled = (state: SignInMethodState) => state !== SignInMethodState.disabled;
+
+export const validateSignInMethods = async (signInMethods: SignInMethods) => {
   const signInMethodStates = Object.values(signInMethods);
   assertThat(
     signInMethodStates.filter((state) => state === SignInMethodState.primary).length === 1,
     'sign_in_experiences.not_one_and_only_one_primary_sign_in_method'
   );
 
-  // TODO: assert others next PR
+  // TODO: LOG-2055 refactor connectors
+  const connectorInstances = await getConnectorInstances();
+  const enabledConnectorInstances = connectorInstances.filter(
+    (instance) => instance.connector.enabled
+  );
```

Comment:
will it be better to keep this method sync - like we use `connectors` as a parameter

Reply:
If some validation fails first, there is maybe no need to get connector instances.
So it's better to get connectors when needed?

Reply:
i would prefer readability > performance, since:
- in most cases connector instances are needed
- this can keep the method "pure"
- refactoring is needed once we'd like to use those instances outside

Reply:
Make sense. Refactored. PTAL.